R/defaultThemes.R

```diff
@@ -139,6 +139,8 @@ initDefaultTheme <- function(){
   theme$subtitle_adj_r <- .9
   theme$legend_intersp_x <- 1
   theme$legend_intersp_y <- 1
+  theme$y_grid_count_strict <- TRUE # do not allow extra ticks to be attached by y_tick_margin (which could lead to y_grid_count being violated)
```

Comment:
does a violation of strict count lead to problem when we have 2 axis? please test.
goes you need to cause a case which violates the count vector and then use 2 axes.

Reply:
data(KOF)
    tt <- initDefaultTheme()
    tt$y_grid_count_strict <- F
    tsplot(KOF$kofbarometer,
           tsr = KOF$reference,auto_legend=F,
           theme = tt)
produces the following plot:
![baro_ref](https://user-images.githubusercontent.com/32675029/35092495-de9a7334-fc3f-11e7-8c25-d3400d79b0e8.png)

The tick lines don't align with the right ticks but I think they are not supposed to anyway? They don't with the default theme either.

Reply:
Hmm, horizontal grids are supposed to align with left and right ticks alike. That's what makes the process a bit complex. I thought they would align, check: 

```
library(tstools)
tsb1 <- ts(runif(30,-30,20),start=c(2010,1),frequency = 4)
tsb2 <- ts(runif(30,0,30),start=c(2010,1),frequency = 4)
tsb3 <- ts(runif(30,0,30),start=c(2010,1),frequency = 4)
tsr1 <- ts(runif(30,-4,6),start=c(2010,1),frequency = 4)
tsplot(tsb1,tsb2,tsb3,tsr=tsr1,
       left_as_bar = F,auto_legend=F)
```

But since it's not the default option, I could live with giving that up when set to FALSE. Because... in practice there won't be a lot of strict = FALSE use cases when there are two axes. Left axis would dominate. Plus we have manual tick often, particularly with stacked bars involved. 

On the other hand fixing this might only be a matter of adding +1 to right axis.
What do you think?